src/hotspot/share/opto/library_call.cpp

```diff
@@ -4318,21 +4318,24 @@ bool LibraryCallKit::inline_array_copyOf(bool is_copyOfRange) {
       klass_node = _gvn.transform(cast);
     }
 
-    // Bail out if either start or end is negative.
+    // Bail out if start is negative.
     generate_negative_guard(start, bailout, &start);
-    generate_negative_guard(end,   bailout, &end);
```

Comment:
I think we need this check to avoid underflow in integer expression (end - start):  (min_int - 1) == max_int

Reply:
OK, I see. Thanks, I readded the `end` guard.